File: 1700-minimum-time-to-make-rope-colorful/minimum-time-to-make-rope-colorful.java
class Solution {
    public int minCost(String colors, int[] neededTime) {
         int n=colors.length();
         char prevchar='1';
         int prevint=1;
        int ans=0;

        for(int i=0;i<n;i++){
                if(prevchar==colors.charAt(i)){
                    if(prevint<=neededTime[i]){
                        ans+=prevint;
                       prevchar=colors.charAt(i);
                       prevint=neededTime[i];
                    }
                    else{
                        ans+=neededTime[i];
                    }
                }
                else{
                   prevchar=colors.charAt(i);
                       prevint=neededTime[i];
                }
        }
        return ans;
    }
}

File: 1700-minimum-time-to-make-rope-colorful/minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& needTime) {
        int ans=0;
        int prev=0,n=colors.size();

        for(int i=1;i<n;i++){
            if(colors[i]==colors[prev]){
                if(needTime[i]>=needTime[prev]){
                    ans+=needTime[prev];
                    prev=i;
                }
                else{
                    ans+=needTime[i];
                }
            }
            else{
                prev=i;
            }
        }

        return ans;
    }
};